src/WebJobs.Script/Models/ExtensionReference.cs

```diff
@@ -1,38 +1,34 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
-using System;
-using System.Collections;
 using System.Collections.Generic;
-using System.Collections.ObjectModel;
-using System.Text;
 
 namespace Microsoft.Azure.WebJobs.Script.Models
 {
     /// <summary>
     /// Represents a binding extension reference.
     /// </summary>
-    public class ExtensionReference
+    public sealed class ExtensionReference
     {
         /// <summary>
-        /// Gets or sets the extension name.
+        /// Gets the extension name.
         /// </summary>
-        public string Name { get; set; }
+        public string Name { get; init; }
 
         /// <summary>
-        /// Gets or sets the assembly-qualified name of the type.
+        /// Gets the assembly-qualified name of the type.
         /// </summary>
-        public string TypeName { get; set; }
+        public string TypeName { get; init; }
 
         /// <summary>
-        /// Gets or sets a hit path that may be used when loading the assembly containing the extension
+        /// Gets a hit path that may be used when loading the assembly containing the extension.
         /// implementation.
         /// </summary>
-        public string HintPath { get; set; }
+        public string HintPath { get; init; }
 
         /// <summary>
-        /// Gets the binding exposed by the extension
+        /// Gets the binding exposed by the extension.
         /// </summary>
-        public ICollection<string> Bindings { get; } = new Collection<string>();
```

Comment:
I’m assuming binding is optional, and this change is likely to have an effect. If the Binding is not defined in the JSON, it will now be null instead of an empty collection. I expect [this ](https://github.com/Azure/azure-functions-host/blob/7e3e7b78fa90ec18eaefd995a9e7797d326d2d00/src/WebJobs.Script/DependencyInjection/ScriptStartupTypeLocator.cs#L163)to result in an exception.
I believe all the tests we've rely on an extension file with binding information, and this scenario slipped through unnoticed.

Reply:
Good find. thank you! There is no easy way to set this to an empty collecting when deserializing using STJ. So I added a null check in the line you linked above before using the `Bindings` property. That is the only place where this property is being used.

The other option is to do make this property `set`table. I don't like that approach since the class is otherwise immutable and would like to keep that way.

Reply:
The null check is to allow executing the code inside the block. I also added a new test which uses extension JSON content without "Bindings" property in it.